In [2]:
import math
import scipy.stats as st
import numpy as np

### 6.4b 
(equal variances)

In [3]:
y_bar1 = 50.3
sd1 = 7.23
var1 = sd1**2
n1 = 13

y_bar2 = 58.6
sd2 = 6.98
var2 = sd2**2
n2 = 21

df = n1 + n2 - 2
print(f"Degrees of freedom: {df}")

alpha = 0.01 / 2  # --> two-tail
crit_val = round(abs(st.t.ppf(alpha, df)), 4)
print(f"Critical value: {crit_val}")

s_pool = round(math.sqrt( ( ( (n1 - 1) * var1) + ( (n2 - 1) * var2)) / df), 4)
print(f"Pooled variance: {s_pool}")

y_diff = round(y_bar1 - y_bar2, 4)
print(f"Mean diff: {y_diff}")

prod = round(crit_val * s_pool * math.sqrt( (1/n1) + (1/n2) ), 4)
print(f"Product: {prod}")

lower = round(y_diff - prod, 4)
upper = round(y_diff + prod, 4)

print(f"{round((1 - (alpha * 2)) * 100, 1)}% confidence interval: ({lower}, {upper})")

Degrees of freedom: 32
Critical value: 2.7385
Pooled variance: 7.0748
Mean diff: -8.3
Product: 6.8373
99.0% confidence interval: (-15.1373, -1.4627)


We are 99% confident that the mean difference between population 1 is 15.1373 to 1.4627 points lower than that of population 2.

### 6.6d 

    Upstream   ave = 4.9067  s = 0.1870        
    Downstream  ave = 3.74    s = 0.3203  
    n = 15
    
(unequal variances) - small sample

In [36]:
y_bar1 = 4.9076
sd1 = 0.1870
n1 = 15

y_bar2 = 3.74
sd2 = 0.3203
n2 = 15

df_num = ((sd1**2 / n1) + (sd2**2 / n2)) ** 2
df_denom = ( ( (sd1**2 / n1) ** 2) / (n1 - 1) ) + ( ( (sd2**2 / n2) ** 2) / (n2 - 1) )

# always round down for degrees of freedom
df = math.floor(df_num / df_denom)
print(f"Degrees of freedom: {df}")

alpha = 0.05 / 2

crit_val = round(abs(st.t.ppf(alpha, df)), 4)
print(f"Critical value: {crit_val}")

avg_diff = round(y_bar1 - y_bar2, 4)
prod = round(math.sqrt( (sd1**2 / n1) + (sd2**2 / n2)), 4)
print(f"Multiplier: {round(prod * crit_val, 4)}")

lower = round(avg_diff - (crit_val * prod), 4)
upper = round(avg_diff + (crit_val * prod), 4)

print(f"{round((1 - (alpha * 2)) * 100, 1)}% confidence interval: ({lower}, {upper})")

Degrees of freedom: 22
Critical value: 2.0739
Multiplier: 0.1987
95.0% confidence interval: (0.9689, 1.3663)


95% confident the average upstream will be 0.9689 and 1.3663 ppm higher than downstream when looking at dissolved oxygen - implying the community dumping sewage into the river is impacting the environment.

### Use 6.9 data to estimate the size of the difference in the mean levels of Magnesium and Europium using a 95% confidence interval 

(unequal variances) - large sample

In [37]:
avg_mag = y_bar1 = 1.0
sd_mag = sd1 = 2.21
n_mag = n1 = 35

avg_eur = y_bar2 = 17
sd_eur = sd2 = 12.65
n_eur = n2 = 35

df_num = ((sd1**2 / n1) + (sd2**2 / n2)) ** 2
df_denom = ( ( (sd1**2 / n1) ** 2) / (n1 - 1) ) + ( ( (sd2**2 / n2) ** 2) / (n2 - 1) )

# always round down for degrees of freedom
df = math.floor(df_num / df_denom)
print(f"Degrees of freedom: {df}")

alpha = 0.05 / 2

crit_val = round(abs(st.t.ppf(alpha, df)), 4)
print(f"Critical value: {crit_val}")

avg_diff = round(y_bar1 - y_bar2, 4)
prod = round(math.sqrt( (sd1**2 / n1) + (sd2**2 / n2)), 4)
print(f"Multiplier: {round(prod * crit_val, 4)}")

lower = round(avg_diff - (crit_val * prod), 4)
upper = round(avg_diff + (crit_val * prod), 4)

print(f"{round((1 - (alpha * 2)) * 100, 1)}% confidence interval: ({lower}, {upper})")

Degrees of freedom: 36
Critical value: 2.0281
Multiplier: 4.4022
95.0% confidence interval: (-20.4022, -11.5978)


We can state we are 95% confident that, of trace metals tracked at the South Pole - the average level of Euriopium is 11.5978 to 20.4022 units higher than that of Magnesium.

### Use 6.23 data to estimate the size of the difference in the mean levels between treatment A and treatment B using a 95% confidence interval

Differences in sample means

In [38]:
t_a = [48.3, 44.6, 49.7, 40.5, 54.3, 55.6, 45.8, 35.4]
t_b = [43.5, 43.8, 53.7, 43.9, 54.4, 54.7, 45.2, 34.4]

t_diff = []
i = 0

alpha = 0.05 / 2

while i < len(t_a):
    t_diff.append(t_a[i] - t_b[i])
    i += 1
# print(t_diff)

y_bar1 = round(np.average(t_a),4)
# print(y_bar1)
y_bar2 = round(np.average(t_b),4)
# print(y_bar2)
y_bar_diff = round(np.average(t_diff),4)
print(f"Difference average: {y_bar_diff}")

# calculated
# print(y_bar_diff)
# estimated
# print(y_bar1 - y_bar2)

n1 = len(t_a)
n2 = len(t_b)
n_diff = len(t_diff)

df_diff = n_diff - 1

var_diff = round( (np.sum(np.square(t_diff)) - ( ( np.sum(t_diff)**2) / n_diff)) / (n_diff - 1), 4)
print(f"Difference variance: {var_diff}")

sd1 = round(float(np.std(t_a)), 4)
# print(sd1)
sd2 =round(float(np.std(t_b)), 4)
# print(sd2)
sd_diff = math.sqrt(var_diff)


crit_val = round(abs(st.t.ppf(alpha, df_diff)), 4)
print(f"Critical value: {crit_val}")


prod = round(math.sqrt( var_diff / n_diff ), 4)
print(f"Multiplier: {round(prod * crit_val, 4)}")

lower = round(y_bar_diff - (crit_val * prod), 4)
upper = round(y_bar_diff + (crit_val * prod), 4)

print(f"{round((1 - (alpha * 2)) * 100, 1)}% confidence interval: ({lower}, {upper})")

Difference average: 0.075
Difference variance: 7.625
Critical value: 2.3646
Multiplier: 2.3086
95.0% confidence interval: (-2.2336, 2.3836)


We can say we are 95% confident that there is $no$ difference between treatments A and B when sampling across eight pairs of twins

### 10.21d
Proportions

In [41]:
new_tox = 5
old_tox = 9

new_nontox = 45
old_nontox = 41

total_tox = new_tox + old_tox
total_nontox = new_nontox + old_nontox
total_new = new_tox + new_nontox
told_old = old_tox + old_nontox

total = total_nontox + total_tox

print(f"Totals: Tox - {total_tox}; Nontox - {total_nontox}; Total - {total}")

Totals: Tox - 14; Nontox - 86; Total - 100


In [43]:
p_old_tox = p_hat1 = round(old_tox / told_old, 4)
p_new_tox = p_hat2 = round(new_tox / total_new, 4)
print(f"P old tox: {p_old_tox}; P new tox: {p_new_tox}")

p_diff = round(p_hat1 - p_hat2, 4)
print(f"P diff: {p_diff}")

alpha = 0.05 / 2
crit_val = abs(round(st.norm.ppf(alpha), 4))
print(f"Critical value: {crit_val}")

prod = round(math.sqrt(((p_hat1 * (1 - p_hat1)) / n1) + ((p_hat2 * (1 - p_hat2)) / n2)), 4)
print(f"Multiplier: {round(prod * crit_val, 4)}")

lower = round(p_diff - (crit_val * prod), 4)
upper = round(p_diff + (crit_val * prod), 4)

print(f"{round((1 - (alpha * 2)) * 100, 1)}% confidence interval: ({lower}, {upper})")

P old tox: 0.18; P new tox: 0.1
P diff: 0.08
Critical value: 1.96
Multiplier: 0.3377
95.0% confidence interval: (-0.2577, 0.4177)


We can conclude that we are 95% confident that there is no difference in proportions of toxic to non-toxic sludge used for agricultural crops when observing the 2 treatment types. 